app/src/main/java/com/example/flightreviewssubmit/ui/recyclerview/viewholder/RateViewHolder.kt

```diff
@@ -0,0 +1,76 @@
+package com.example.flightreviewssubmit.ui.recyclerview.viewholder
+
+import android.view.View
+import android.widget.RatingBar
+import androidx.appcompat.widget.AppCompatRatingBar
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.flightreviewssubmit.R
+import com.example.flightreviewssubmit.data.RateFlightData
+import com.example.flightreviewssubmit.ui.recyclerview.adapter.FlightSubmitAdapter
+import java.lang.UnsupportedOperationException
+
+sealed class RateViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+    abstract fun bind(
+        data: RateFlightData,
+        action: FlightSubmitAdapter.IRateActionListener,
+        isActive: Boolean)
+
+    class RateCrowdedViewHolder(itemView: View) : RateViewHolder(itemView){
+        private val crowdRateBar: AppCompatRatingBar = itemView.findViewById(R.id.crowd_rate_bar)
+
+        override fun bind(
+            data: RateFlightData,
+            action: FlightSubmitAdapter.IRateActionListener,
+            isActive: Boolean
+        ) {
+            if (data !is RateFlightData.RateCrowd)
+                throw UnsupportedOperationException("Unsupported data. " +
+                        "Data have to be type: RateFlightData.RateFlight")
+
+            crowdRateBar.isEnabled = isActive
+
+            // -1 cause rating range(1,6)
+            crowdRateBar.rating = data.rating.value.toFloat() - 1
+
+            crowdRateBar.onRatingBarChangeListener = RatingBar.OnRatingBarChangeListener {
+                    _, rating, _ ->
+                        // +1 cause rating range(1,6)
+                        data.rating.value = rating.toInt() + 1
+                        action.setRating(data)
+            }
```

Comment:
Creating a new listener for each `bind` call could be bad for performance of long lists. We should create and set all the listeners once in an `init` block (or via `.apply` directly on class fields)